### DISCUSSION TITLE
:rocket: 테스트 API 서버 - `종합본`

### DISCUSSION THREAD

**Discussion by kimminkyeu** [Kernel360/F2-BAGUNI#196] [메모장]
프록시, SSL 설정은 `nginx-proxy` 와 `acme-companion` 를 사용해 설정 했습니다.
- 링크 : https://github.com/nginx-proxy

<img src="https://github.com/user-attachments/assets/fe77f904-03ef-4918-8b0c-5916fd1b58a6" width="70%">

### :link: [minlife.me](https://minlife.me/) - `관리자 페이지`
- 현재는 `portainer`만 연결했기에, 접속시 컨테이너 관리 앱으로 갑니다.
관리자 접속 아이디, 암호는 [노션 페이지 최상단 댓글창](https://www.notion.so/Tech-Pick-2-a8762c3a0c3b48c998376cde063a2d9b?d=5cc6dc899f424d908a5f356da9157407&pvs=4)에 있습니다.
- :construction: 추후 Admin 페이지 (ex. 통계 확인)가 추가될 경우 이 URL에 연결할 예정

### :link:  [v1.minlife.me](https://v2.minlife.me/) 
- `backend-v1` 브랜치와 github action ci/cd 연결
- 기존 api + 데이터베이스 = `techpick_db_v1`
- api 명세 swagger : :link: [v1.minlife.me/swagger-ui](https://v1.minlife.me/swagger-ui/index.html)

### :link:  [v2.minlife.me](https://v2.minlife.me/)
- `backend-v2` 브랜치와 github action ci/cd 연결
- 데이터베이스 = `techpick_db_v2`
- api 명세 swagger : :link: [v2.minlife.me/swagger-ui](https://v2.minlife.me/swagger-ui/index.html)

### :computer: SSH 터미널 접속법
```bash
ssh {접속 계정}@minlife.me -p 4242     # 도메인으로 접속
ssh {접속 계정}@182.224.90.172 -p 4242     # 고정 IP로 접속
```
접속 계정 추가는 @kimminkyeu 에게 문의할 것.

### :tada: API 사용법
1. 각 API 버전에 맞는 URL에 접속, 로그인해서 쿠키를 획득.
2. 브라우저 개발자도구 --> `access_token` 쿠키 값이 있음을 확인.
3. swagger 사이트 접속

## :arrow_down: 홈 서버 NAT 설정 방식
아래와 같이 포트 매핑 되어 있습니다. 
집에 있는 U+ 홈 게이트웨이 거치고 -->  U+ 무선 공유기를 거친 뒤 --> 제 방에 있는 서버 노트북과 연결됩니다.

![image](https://github.com/user-attachments/assets/8ad73a0a-9b93-46ed-a6ee-8fc60560f717)

**Comment by Gyaak** [comment]
이건 wiki로 옮겨주시면 더 좋을것같습니다.

**Reply by kimminkyeu** [reply]
네~ 상원님까지 확인 되시면 위키로 이전할께요

**Comment by sangwonsheep** [comment]
확인 완료했습니다! 위키로 옮겨주시면 감사하겠습니다!
